appserver/web/web-core/src/main/java/org/apache/catalina/fileupload/PartItem.java

```diff
@@ -733,10 +734,30 @@ private void readObject(ObjectInputStream in)
         // read values
         in.defaultReadObject();
 
+        if (repository != null) {
+            if (repository.isDirectory()) {
+                                // Check path for nulls
+                if (repository.getPath() == null || repository.getPath().contains("\0")) {
```

Comment:
This doesn't seem right. The isDirectory check has succeeded at this point, so getPath() can't be null. You can quickly check by running FindBugs.